asdf/generic_io.py

```diff
@@ -779,11 +779,9 @@ def read_into_array(self, size):
         return np.fromfile(self._fd, dtype=np.uint8, count=size)
 
     def close(self):
+        self.flush_memmap()
         super().close()
-        if self._close:
-            if hasattr(self, "_mmap") and not self._mmap.closed:
-                self._mmap.flush()
-                self._mmap.close()
```

Comment:
Are there any additional implications of this change?

Reply:
None that I've been able to uncover. It does not appear that any other tests aside from the one added in this PR and the ones in astropy (that load a table from asdf) suffer from this issue. With 2.13 memmaps were handled by numpy.memmap which has no api for closing the memmap (as per the docstring here: https://github.com/numpy/numpy/blob/v1.23.0/numpy/core/memmap.py#L40-L43) so it's difficult if not impossible to say we've mimicked the old behavior.
Is there something specific that you had in mind?

Reply:
No, it seems like a reasonable change then.